jvector-base/src/main/java/io/github/jbellis/jvector/graph/ConcurrentNeighborSet.java

```diff
@@ -119,10 +125,10 @@ public void replaceDeletedNeighbors(Bits deletedNodes, NodeArray candidates) {
                 }
             }
 
-            // merge the live neighbors with the candidates and prune down to diverse edges
-            var merged = mergeNeighbors(liveNeighbors, candidates);
-            var newNeighbors = copyDiverse(merged, selectDiverse(merged, 0));
-            return new Neighbors(newNeighbors, newNeighbors.size);
+            // merge the remaining neighbors with the candidates
+            NodeArray merged = rescoreAndMerge(liveNeighbors, candidates);
```

Comment:
yes, I went ahead and changed this code in advance of implementing

                // TODO support approximate scoring here

in removeDeletedNodes, because I thought it was likely I'd forget

Reply:
... actually now that we are using the FreshDiskANN algorithm I don't think it makes sense to use approximate scoring at all.  will update